### ISSUE TITLE
pd2bot cow problem 

### ISSUE THREAD

**Issue by sadinsane** [kolton/d2bot-with-kolbot#3003]
i dont know why when bot try to do cow i always got that error 
Error in Cows (pather.js #602) moveToPreset: Couldn't find preset unit - id 888 (Area: 4,
cain quest is done baal quest too before my other bot got banned i was able to do cow easy now i cant wtih that error

**Comment by gtoilet** [comment]
so what does it do exactly?
does it go get leg/tome , try to make cows and fails?
or does it get into cows and then cant find unit id?

**Comment by sadinsane** [comment]
> 
> 
> so what does it do exactly?
> does it go get leg/tome , try to make cows and fails?
> or does it get into cows and then cant find unit id?

it goes to stony field and its like it cant find the portal for tristram she stay at the wp

**Comment by gtoilet** [comment]
when she goes to wp
pause bot and try to manually walk toward trist portal
unpause and see if she still afks or goes in

try to determine if the pathing to the trist port is the problem
since that server has mods and such they MAY have modified how the bot pathers to the portal ... idk as dont play that server

**Comment by sadinsane** [comment]
> 
> 
> when she goes to wp
> pause bot and try to manually walk toward trist portal
> unpause and see if she still afks or goes in
> 
> try to determine if the pathing to the trist port is the problem
> since that server has mods and such they MAY have modified how the bot pathers to the portal ... idk as dont play that server

kinda hard to pause at wp right when she get in stony field she came back town

**Comment by gtoilet** [comment]
maby a delay after taking wp will help
no idea what could be the problem with that server and its mods
on another thread i helped a guy remove the is cow king killed error
that would still let you make cow portal even after king was killed on that toon

**Comment by sadinsane** [comment]
> 
> 
> maby a delay after taking wp will help
> no idea what could be the problem with that server and its mods
> on another thread i helped a guy remove the is cow king killed error
> that would still let you make cow portal even after king was killed on that toon

how can i delay after taking wp ?

**Comment by gtoilet** [comment]
figure you use the stock cows.js
on line 81 deals with getting the leg
on line 88 it uses wp 4 so delay after that long enuf for you to  pause the script before it goes back to town so that you can move toward the trist portal and see if it then takes it , should look something like this


	this.getLeg = function () {
		var i, portal, wirt, leg, gid;

		if (me.getItem(88)) {
			return me.getItem(88);
		}

		Pather.useWaypoint(4);
			delay(500);
		Precast.doPrecast(true);

**Comment by gtoilet** [comment]
also in that this.getLeg
if that server has infact done something to effect the pathing to trist
that is the function you need to edit to get it working

**Comment by sadinsane** [comment]
> 
> 
> also in that this.getLeg
> if that server has infact done something to effect the pathing to trist
> that is the function you need to edit to get it working

yeah it seem they change some thing to prevent bot to find trist portal cuz i cant do rakanichu too and and cant do trist too
